lenskit-eval/src/main/java/org/lenskit/eval/traintest/ExperimentJob.java

```diff
@@ -154,43 +155,48 @@ protected void compute() {
                                                     .setCount(nusers)
                                                     .setLabel("testing users")
                                                     .start();
-            // TODO Support something other than ratings
-            for (Entity user: testData.query(CommonTypes.USER).get()) {
-                if (Thread.interrupted()) {
-                    throw new EvaluationException("eval job interrupted");
-                }
-                long uid = user.getId();
-                userRow.add("User", uid);
 
-                List<Rating> userTrainHistory = trainData.query(Rating.class)
-                                                         .withAttribute(CommonAttributes.USER_ID, uid)
-                                                         .get();
-                List<Rating> userTestHistory = testData.query(Rating.class)
-                                                       .withAttribute(CommonAttributes.USER_ID, uid)
-                                                       .get();
-                TestUser testUser = new TestUser(user, userTrainHistory, userTestHistory);
+            List<EntityType> entityTypes = dataSet.getEntityTypes();
 
-                Stopwatch userTimer = Stopwatch.createStarted();
+            for (EntityType entityType: entityTypes) {
```

Comment:
We don't want to wrap the whole evaluation loop in a new loop - we only need to loop over `entityTypes` for the purpose of filling the `userTrainHistory` and `userTestHistory` variables.